syntax = "proto3";

service PythonGrpcService {
  rpc SendCode(GrpcPythonRequest) returns (GrpcPythonResponse) {}

  rpc SendPip(GrpcPythonPipRequest) returns (GrpcPythonPipResponse) {}
}

message GrpcPythonRequest {
  string script = 1;
  repeated string fieldNames = 2;
}

message GrpcPythonResponse {
  map<string, string> fields = 1;
}

message GrpcPythonPipRequest {
  string name = 1;
  string libraryName = 2;
  repeated string options = 3;
}

message GrpcPythonPipResponse {
  bool successful = 1;
}